test/e2e-samples/page-objects/authenticators-page.js

```diff
@@ -12,31 +12,26 @@
 
 'use strict';
 
-const util = require('./shared/util');
-
 class AuthenticatorsPage {
-
-  constructor() {
-    this.authenticatorLabelSelector = $$('.authenticator-label');
-    this.selectAuthenticatorButton = $$('[class="authenticator-button"] [data-se="button"]');
-  }
+  
+  get authenticatorLabelSelector () { return $$('.authenticator-label'); }
+  get selectAuthenticatorButton () { return $('[class="authenticator-button"] [data-se="button"]'); }
 
   waitForPageLoad() {
-    return util.wait(this.selectAuthenticatorButton);
+    return this.selectAuthenticatorButton.waitForDisplayed();
   }
 
-  clickAuthenticatorByLabel(label) {
-    util.wait(this.authenticatorLabelSelector);
-
-    this.authenticatorLabelSelector.each((element, index) => {
-      element.getText().then((text) => {
-        if (text.includes(label) > 0) {
-          this.selectAuthenticatorButton.get(index).click();
-        }
-      });
+  async clickAuthenticatorByLabel(label) {
+    await this.authenticatorLabelSelector.waitForDisplayed();
 
+    // TODO: verify this
+    this.authenticatorLabelSelector.each(async (element, index) => {
```

Comment:
use `for ... of ` instead of `each` function, I remember `each` does not work well with async functions.